plugins/gradients/index.js

```diff
@@ -0,0 +1,17 @@
+const plugin = require('tailwindcss/plugin')
+
+module.exports = plugin(({ addUtilities, e, theme, variants }) => {
+  const pluginConfig = theme('gradients', {})
+  const pluginVariants = variants('gradients', [])
+
+  const utilities = Object.keys(pluginConfig).map(([key, value]) => {
+    return {
+      [`.gradient-${key}`]: {
```

Comment:
Might also want to use the `e` method to escape the class names, just incase someone tries to name the keys with fractions, e.g. `gradient-blue-1/2`. Actually I think the docs recommend always wrapping user provided selectors this way as a safeguard.

Reply:
https://tailwindcss.com/docs/plugins/#escaping-class-names